File: 品味Java的21種設計模式/dp_cpp/prototype/example2/ConcretePrototype2.cpp
#include "ConcretePrototype2.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace prototype
			{
				namespace example2
				{

					Prototype *ConcretePrototype2::clone()
					{
						//最简单的克隆，新建一个自身对象，由于没有属性，就不去复制值了
						Prototype *prototype = new ConcretePrototype2();
						return prototype;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/prototype/example2/ConcretePrototype2.java
package cn.javass.dp.prototype.example2;

/**
 * 克隆的具体实现对象
 */
public class ConcretePrototype2 implements Prototype {
    public Prototype clone() {
        //最简单的克隆，新建一个自身对象，由于没有属性，就不去复制值了
        Prototype prototype = new ConcretePrototype2();
        return prototype;
    }
}